libpod/runtime_ctr.go

```diff
@@ -687,11 +687,17 @@ func (r *Runtime) removeContainer(ctx context.Context, c *Container, force, remo
 	}
 
 	if c.IsService() {
-		canStop, err := c.canStopServiceContainer()
-		if err != nil {
-			return err
+		var pods []string
+		for _, id := range c.state.Service.Pods {
```

Comment:
searching the code I cannot find any place where you remove a pod from this list?
If a pod is removed it must be removed from this list. Otherwise podman could reuse the old pod id for a new pod which will then be part of this service which looks wrong. I know it is very unlikely but it is also very hard to debug if it ever happens in the wild.

Reply:
Good thinking.  I'll do that.